### PR TITLE
Fix turrets immediately moving from map-defined facing to InitialFacing.

### LINKED ISSUE #18635: Turret facing applied in editor not displayed ingame.
Tested in the openra bleed version compiled half an hour ago, i edited the turret facing of the nod turrets in the shellmap.

image

Meanwhile in the editor

image

### PR DESCRIPTION
**Pull request by pchote**
Fixes #18635.

The facings actually are displayed, but the turret realignment logic immediately moves them back to the mod-defined default facing. This happens because #18418 basically changed the turret logic from "keep the current world facing unless something tells you otherwise" to "realign to the default relative facing unless something tells you otherwise".

This PR ~tells the turret otherwise, and it should hold the initial facing override until the realign timer expires or another trait requests something else. `WVec.Zero` is an abuse of implementation details, but saves otherwise unnecessary vector calculations each tick.~ now restores this original behaviour.

### CODE REVIEW

**Comment by pchote** [comment]
It would ideally be cleaner if we could change the default logic back to how it was, with the behaviour for "no desired facing" being to not turn, and the behaviour on the realign timer expiry being to request it faces the default direction. We can't do this because the facing requests are given as a direction in the world, without a way to specify a specific local facing.

**Comment by ABrandau** [comment]
Solves the issue, turrets start showing their own original facing (each variant of the crate is a facing out of itself)
unknown

**Comment by pchote** [comment]
Updated as discussed on Discord.

File: OpenRA.Mods.Common/Traits/Turreted.cs [hunk 1]
```diff
@@ -136,7 +136,8 @@ public class Turreted : PausableConditionalTrait<TurretedInfo>, ITick, IDeathAct
 		[Sync]
 		public int QuantizedFacings = 0;
 
-		WVec? desiredDirection = null;
+		bool desiredRealign;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Turreted.cs]: `realignDesired` sounds more natural to me.

**Comment by pchote** [comment]
Updated.

**Review by abcdefg30 — APPROVED** [review]

**Comment by reaperrr** [comment]
I set different map-defined facings on some allied base turrets and soviet heavy tanks on the RA shellmap on this branch, and they still immediately realigned. Am I missing something, or is it just not working as advertised?

**Comment by pchote** [comment]
The expected behaviour is for them to realign after `RealignDelay` ticks, instead of 0 ticks. The default `RealignDelay` in RA is pretty short, so you may not notice the difference unless you increase this to a larger number.

**Comment by reaperrr** [comment]
OK, I see.
Only indirectly related to this PR then, but do we really want defense turrets to realign as well though? It looks really weird when the gun turrets always turn back to their default west facing.

**Comment by pchote** [comment]
I agree. Lets fix that in a followup PR.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/0bdd9897cc799916ccf31a9857afd0876e306bbb...665ef66dddaab5853b0bacec07004d4fc81dbd9b

**Pull request merged**